scripts/setup/install.ps1

```diff
@@ -42,6 +42,48 @@ Param(
 
 $_rollbackStore = @{}
 
+function Retry-Command
+{
+    param (
+    [Parameter(Mandatory=$true)][string]$command,
+    [Parameter(Mandatory=$true)][hashtable]$args,
+    [Parameter(Mandatory=$false)][int]$retries = 2,
+    [Parameter(Mandatory=$false)][int]$secondsDelay = 1,
+    [Parameter(Mandatory=$false)][switch]$throwError
+    )
+
+    # Setting ErrorAction to Stop is important. This ensures any errors that occur in the command are
+    # treated as terminating errors, and will be caught by the catch block.
+    $args.ErrorAction = "Stop"
+
+    $retrycount = 0
+    $completed = $false
+
+    while (-not $completed) {
+        try {
+            & $command @args
+            Write-Verbose ("Command {0} succeeded." -f $command)
+            $completed = $true
+
+        } catch {
```

Comment:
We shouldn't blindly retry `Start-Service`. A better thing to do is to retry `Get-Service` to determine if the service is ready

Reply:
Repeating Start-Service should be okay. If the service is running, it is No operation.

Reply:
I disagree. If there's a critical issue we shouldn't even waste time retrying. This would actually echo my comments from earlier allow us to increase retry period if service took longer to create. Just because we can only produce a scenario where retry after 1 second is sufficient doesn't mean it would always be the case